arrow-site/docs/docs/patterns/monads/README.md

```diff
@@ -365,318 +348,239 @@ Subsequently, `addStep` is called two times to transfer to `Conference` and then
 
 The name of this pattern is **Monad**.
 
-In Arrow terms, a Monad is an interface with two operations: A constructor `just`, and `flatMap`.
+In Arrow terms, a `Monad` can be formalized by the powers provided in an impl of the `Effect<F>` interface.
+The `Effect` interface defines the ability to have a delimited scope in which a continuation can be used to complete the context or yield a value.
 
 ```kotlin
-interface Monad<F>: Applicative<F>, Functor<F>, Selective<F> {
-    fun <A> just (instance: A): Kind<F, A> // this comes from [Applicative] interface
-
-    fun <A, B> Kind<F, A>.flatMap(f: (A) ->  Kind<F, B>) : Kind<F, B>
+fun interface Effect<F> {
+  fun control(): DelimitedScope<F>
 }
 ```
 
-See that, instead of `WorkflowThatReturns<A>`, our containers are called `Kind<F, A>`, where `F` is the generic parameter of the container and `A` the generic parameter of the content. We talk more about them in the [glossary]({{ '/patterns/glossary/#type-constructors' | relative_url }}).
+We can then define specific effects for our data types that implement Monad bind.
 
-The constructor `just` is used to put an object into a container `Kind<F, A>`, and `flatMap` is used to replace one contained object with another contained object.
+```kotlin:ank
+import arrow.continuations.Effect
 
-We can think of `flatMap` as a combination of `map` and `flatten`, as defined by the following signature:
+fun interface NullableEffect<A> : Effect<A?> {
+  suspend fun <B> B?.bind(): B =
+    this ?: control().shift(null)
+}
+```
 
-```kotlin
-fun <F, A, B> Kind<F, A>.map(f: (A) ->  B): Kind<F, B> // Inherited from Functor
+Monad `bind` is then implemented by shifting out of the context with `null` if the value being bound happens to be null or yielding the value by returning it if not null.
+With `control().shift` based on the `Effect` interface we can implement monad bind short-circuiting and other patterns supported by the Continuation monad for arbitrary data types.
```

Comment:
I'd show at least what `DelimitedScope` is before introducing `shift`, otherwise there's a jump there.